File: features/offices/src/main/java/com/nivasafinance/features/offices/service/OfficeWriteService.java
package com.nivasafinance.features.offices.service;

import com.nivasafinance.features.offices.dto.OfficeCreateRequest;
import com.nivasafinance.features.offices.dto.OfficeResponse;

public interface OfficeWriteService {
    OfficeResponse createOffice(OfficeCreateRequest request);

    OfficeResponse moveOffice(String officeKey, String newParentKey);

    OfficeResponse activateOffice(String officeKey);

    OfficeResponse deactivateOfficeCascade(String officeKey);
}

File: features/offices/src/main/java/com/nivasafinance/features/offices/service/OfficeWriteService.kt
package com.nivasafinance.features.offices.service

import com.nivasafinance.features.offices.dto.OfficeCreateRequest
import com.nivasafinance.features.offices.dto.OfficeResponse

interface OfficeWriteService {
    fun createOffice(request: OfficeCreateRequest): OfficeResponse
}
